File: truss-compiler/src/main/java/truss/compiler/symbols/ConstructorSymbol.java
package truss.compiler.symbols;

public class ConstructorSymbol extends MethodSymbol {
    @Override
    public SymbolKind getKind() {
        return SymbolKind.CONSTRUCTOR;
    }

    @Override
    public void accept(SymbolVisitor visitor) {
        if (!visitor.isDone()) {
            visitor.visitConstructor(this);
        }
    }
}

File: Truss.Compiler/Symbols/ConstructorSymbol.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.Symbols {
    public class ConstructorSymbol : MethodSymbol {
        public override SymbolKind Kind {
            get { return SymbolKind.Constructor; }
        }

        public override void Accept(ISymbolVisitor visitor) {
            if (!visitor.Done) {
                visitor.VisitConstructor(this);
            }
        }
    }
}
